docs/source/user_guide/types.rst

```diff
@@ -0,0 +1,103 @@
+======================
+Type Support In Koalas
+======================
+
+.. currentmodule:: databricks.koalas
+
+In this chapter, we will briefly show you how data types change when converting Koalas DataFrame from/to PySpark DataFrame or pandas DataFrame.
+
+
+Type casting between PySpark and Koalas
+---------------------------------------
+
+When converting the Koalas DataFrame from/to PySpark DataFrame, the data types automatically casted to the appropriate type.
+
+The example below shows how types are casted between PySpark DataFrame and Koalas DataFrame.
+
+.. code-block:: python
+
+    # 1. Create PySpark DataFrame
+    >>> sdf = spark.createDataFrame([
+    ...     (1, Decimal(1.0), 1., 1., 1, 1, 1, datetime(2020, 10, 27), "1", True),
+    ... ], 'tinyint tinyint, decimal decimal, float float, double double, integer integer, long long, short short, timestamp timestamp, string string, boolean boolean')
+
+    # 2. Check the PySpark data types
+    >>> sdf
+    DataFrame[tinyint: tinyint, decimal: decimal(10,0), float: float, double: double, integer: int, long: bigint, short: smallint, timestamp: timestamp, string: string]
+
+    # 3. Convert PySpark DataFrame to Koalas DataFrame
+    >>> kdf = sdf.to_koalas()
+
+    # 4. Check the Koalas data types
+    >>> kdf.dtypes
+    tinyint                int8
+    decimal              object
+    float               float32
+    double              float64
+    integer               int32
+    long                  int64
+    short                 int16
+    timestamp    datetime64[ns]
+    string               object
+    boolean                bool
+    dtype: object
+
+    # 5. Easily go back to the PySpark DataFrame
+    >>> sdf = kdf.to_spark()
+
+    # 6. Check the PySpark data types again
+    >>> sdf
+    DataFrame[tinyint: tinyint, decimal: decimal(10,0), float: float, double: double, integer: int, long: bigint, short: smallint, timestamp: timestamp, string: string]
+
+
+Type casting between pandas and Koalas
+--------------------------------------
+
+We can easily convert Koalas DataFrame to pandas DataFrame, and the data types are basically same as pandas.
+
+.. code-block:: python
+
+    # Convert Koalas DataFrame to pandas DataFrame
+    >>> pdf = kdf.to_pandas()
+
+    # Check the pandas data types
+    >>> pdf.dtypes
```

Comment:
The document looks pretty clear to me! nit: Within the `Type casting between pandas and Koalas` section itself, we may not tell that column names indicate respective PySpark types. I'm not sure whether we want to point this out, just FYI.

Reply:
Thanks for the opinion, @xinrong-databricks ! Let me check about this :)